scripts/lib/api/Pkg.ts

```diff
@@ -20,35 +30,156 @@ export interface Link {
+
+  constructor(kwargs: {
+    name: string;
+    title: string;
+    githubSlug: string;
+    hasSeparateReleaseNotes: boolean;
+    transformLink?: (link: Link) => Link | undefined;
+    version: string;
+    versionWithoutPatch: string;
+    historical: boolean;
+    releaseNoteEntries: ReleaseNoteEntry[];
+  }) {
+    this.name = kwargs.name;
+    this.title = kwargs.title;
+    this.githubSlug = kwargs.githubSlug;
+    this.hasSeparateReleaseNotes = kwargs.hasSeparateReleaseNotes;
+    this.transformLink = kwargs.transformLink;
+    this.version = kwargs.version;
+    this.versionWithoutPatch = kwargs.versionWithoutPatch;
+    this.historical = kwargs.historical;
+    this.releaseNoteEntries = kwargs.releaseNoteEntries;
+  }
+
+  static async fromArgs(
+    name: string,
+    version: string,
+    versionWithoutPatch: string,
+    historical: boolean,
+  ): Promise<Pkg> {
+    const args = {
+      name,
+      version,
+      versionWithoutPatch,
+      historical,
+    };
+
+    if (name === "qiskit") {
+      const releaseNoteEntries = await findLegacyReleaseNotes(name);
+      return new Pkg({
+        ...args,
+        title: "Qiskit",
+        name: "qiskit",
+        githubSlug: "qiskit/qiskit",
+        hasSeparateReleaseNotes: true,
+        releaseNoteEntries,
+      });
+    }
+
+    if (name === "qiskit-ibm-runtime") {
+      return new Pkg({
+        ...args,
+        title: "Qiskit Runtime IBM Client",
+        name: "qiskit-ibm-runtime",
+        githubSlug: "qiskit/qiskit-ibm-runtime",
+        transformLink,
+        hasSeparateReleaseNotes: false,
+        releaseNoteEntries: [],
+      });
+    }
+
+    if (name === "qiskit-ibm-provider") {
+      return new Pkg({
+        ...args,
+        title: "Qiskit IBM Provider",
+        name: "qiskit-ibm-provider",
+        githubSlug: "qiskit/qiskit-ibm-provider",
+        transformLink,
+        hasSeparateReleaseNotes: false,
+        releaseNoteEntries: [],
+      });
+    }
+
+    throw new Error(`Unrecognized package: ${name}`);
+  }
+
+  static mock(kwargs: {
+    name?: string;
+    title?: string;
+    githubSlug?: string;
+    hasSeparateReleaseNotes?: boolean;
+    transformLink?: (link: Link) => Link | undefined;
+    version?: string;
+    versionWithoutPatch?: string;
+    historical?: boolean;
+    releaseNoteEntries?: ReleaseNoteEntry[];
+  }): Pkg {
+    return new Pkg({
+      name: kwargs.name ?? "my-quantum-project",
+      title: kwargs.title ?? "My Quantum Project",
+      githubSlug: kwargs.githubSlug ?? "qiskit/my-quantum-project",
+      hasSeparateReleaseNotes: kwargs.hasSeparateReleaseNotes ?? false,
+      transformLink: kwargs.transformLink,
+      version: kwargs.version ?? "0.1.0",
+      versionWithoutPatch: kwargs.versionWithoutPatch ?? "0.1",
+      historical: kwargs.historical ?? false,
+      releaseNoteEntries: kwargs.releaseNoteEntries ?? [],
+    });
+  }
+
+  outputDir(parentDir: string): string {
```

Comment:
I got rid of the default arg `parentDir = "docs"`. It wasn't used and imo the default was dangerous because you might fail to realize you need to set it.